### ISSUE TITLE
how to load the python tenserflow model to java tensorflow and predict new image

### ISSUE THREAD

**Issue by chaowenGUO** [tensorflow/java#403]
Hi:

I use the following code to construct a mnist digit model and save it to model
import tensorflow, numpy
(x_train, y_train), (x_test, y_test) = tensorflow.keras.datasets.mnist.load_data()
x_train, x_test = x_train / 255, x_test / 255
model = tensorflow.keras.models.Sequential([
  tensorflow.keras.layers.Flatten(input_shape=x_train[0].shape),
  tensorflow.keras.layers.Dense(128, activation='relu'),
  tensorflow.keras.layers.Dropout(0.2),
  tensorflow.keras.layers.Dense(10, activation='softmax')
])
model.compile(optimizer='adam',
              loss='sparse_categorical_crossentropy',
              metrics=['accuracy'])
model.fit(x_train, y_train, epochs=5)
model.save('model')

In python, I can use:
probability_model = tensorflow.keras.Sequential([tensorflow.keras.models.load_model('model'), tensorflow.keras.layers.Softmax()])
predictions = probability_model.predict(numpy.array([x_test[0]]))
[numpy.argmax(_) for _ in predictions]
to do prediction

Now I need to load my python model to java tensorflow, so I need to how to load the model in java. But I can not find any document or example about how to do that. Please give me an example how to do prediction in java.

**Comment by Craigacp** [comment]
By default Keras saves models in TF's `SavedModel` format. That can be loaded using our `SavedModelBundle` class, which loads in the functions defined in the model. You can then call those functions and supply them the input. I'd expect your saved model to already have a softmax on the top given the way it's defined, but if not you can pass it through our Softmax op to get the same effect.

**Comment by karllessard** [comment]
@chaowenGUO , I'm revisiting old issues, did you figured out how to load the saved model in Java? Basically, like @Craigacp  explained, you should load it as a [`SavedModelBundle`](https://github.com/tensorflow/java/blob/master/tensorflow-core/tensorflow-core-api/src/main/java/org/tensorflow/SavedModelBundle.java) instance and you can directly run inference on it. You should be careful with resource scoping though, to make sure native memory can be freed properly

For example:
```java
try (SavedModelBundle model = SavedModelBundle.load("path_to_your_saved_model_dir")) {
    try (TFloat32 x = TFloat32.vectorOf(0.5f, 0.4f, 0.3f);
         Result r = model.call(Map.of("x", x))) {
        System.out.println(r.get(0).getFloat());
    }
}
```

**Comment by karllessard** [comment]
I've also answered to that same question awhile ago on SO, you can take a look at [here](https://stackoverflow.com/questions/62241546/java-tensorflow-keras-equivalent-of-model-predict/62295153#62295153)
